### ISSUE TITLE
Feature request: allow multiple expressions in a single program

### ISSUE THREAD

**Issue by vmihailenco** [expr-lang/expr#697]
TLDR The following program should be a valid expr:

```
let x = 42
let y = 42
x * y
```

It should work like Ruby, where everything is an expression and the last expression is used as the result. Given that it works fairly well for Ruby, I guess expr-lang can be modeled just like it.

Currently what we do in [Uptrace transformations](https://uptrace.dev/get/transformations.html#script), is just split the multi-line expressions into individual programs and then run them one by one. It works so far, but variables don't survive between programs and the performance is probably sub-optimal since each program requires a separate `VM`.

I believe OpenTelemetry does the same with multi-line [OTTL](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/ottl/README.md) so they would probably benefit as well.

A more realistic example what multi-line expressions can achieve:

```
let val = split(attr("foo"), "-")
setAttr("xxx", val[0])
setAttr("yyy", val[1])
deleteAttr("foo")
```

**Comment by antonmedv** [comment]
Well, actually this is possible right now as well if `;` added at the let ends:

```js
let x = 42;
let y = 42;
x * y
```

We can improve lexer and allow `\n` to be treated as `;`.

**Comment by vmihailenco** [comment]
>Well, actually this is possible right now as well if ; added at the let ends:

Yes, I know, but the real program looks more like this:

```
let val = split(attr("foo"), "-")
setAttr("xxx", val[0])
setAttr("yyy", val[1])
deleteAttr("foo")
```

I guess I can rewrite it like this, but it is not as readable and debuggable:

```
let val = split(attr("foo"), "-"); setAttr("xxx", val[0]) && setAttr("yyy", val[1]) && deleteAttr("foo")
```

**Comment by antonmedv** [comment]
I see. So, this request to add statements to Expr. I think this is nice to add via option: `expr.AllowStatments()`.

**Comment by vmihailenco** [comment]
>So, this request to add statements to Expr.

Maybe, but I explicitly say that "everything is an expression" so :) If we take [switch](https://github.com/expr-lang/expr/issues/586) in account,  would it be a statement or expression? I believe it should be an expression or it probably would not find into the purpose of the language.

**Comment by xiaoas** [comment]
Statements support(or using `;` to simply discard expression output) would be very helpful indeed. I'm currently having the following usecase:
```js
assert(status_code == 0);
assert(status_message == "");
// normal expr stuff
1 + 1
```

Currently it's possible to work around this by
1. Making `assert` return an `any`.
```diff
- func assert(condition bool)
+ func assert(condition bool) any // always returns nil
```
2. change the assert statement to assignments:
```js
let _s1 = assert(status_code == 0);
let _s2 = assert(status_message == "");
// normal expr stuff
1 + 1
```

**Comment by Alex-Young1113** [comment]
It seems that the only case that multiple statement is allowed is multiple variable definition like:
```
let z = x + y;
let w = x - y;
z * w
```

If I want to simply make multiple function calls it will fail.
```go
func TestExprCustomFunction(t *testing.T) {
	env := map[string]any{
		"greet":   "Hello, %v!",
		"names":   []string{"world", "you"},
		"sprintf": fmt.Sprintf,
	}
	e, err := expr.Compile(`sprintf(greet, names[0]); sprintf(greet)`, expr.Env(env))
	if err != nil {
		panic(err)
	}

	out, err := expr.Run(e, env)
	if err != nil {
		panic(err)
	}

	fmt.Println(out)
	fmt.Println(reflect.TypeOf(out))
}
```

**Comment by antonmedv** [comment]
I’m working on adding multi-expressions available.
